src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -0,0 +1,462 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // Contains test methods that can be used for FileInfo, DirectoryInfo, File or Directory.
+    public abstract class BaseSymbolicLinks_FileSystem : BaseSymbolicLinks
+    {
+        private const string ExtendedPrefix = @"\\?\";
+        /// <summary>Creates a new file or directory depending on the implementing class.
+        /// If createOpposite is true, creates a directory if the implementing class is for File or FileInfo, or
+        /// creates a file if the implementing class is for Directory or DirectoryInfo.</summary>
+        protected abstract void CreateFileOrDirectory(string path, bool createOpposite = false);
+        protected abstract void AssertIsCorrectTypeAndDirectoryAttribute(FileSystemInfo linkInfo);
+        protected abstract void AssertLinkExists(FileSystemInfo linkInfo);
+        /// <summary>Calls the actual public API for creating a symbolic link.</summary>
+        protected abstract FileSystemInfo CreateSymbolicLink(string path, string pathToTarget);
+        /// <summary>Calls the actual public API for resolving the symbolic link target.</summary>
+        protected abstract FileSystemInfo ResolveLinkTarget(string linkPath, bool returnFinalTarget = false);
+
+        /// <summary>
+        /// Verifies that FileSystemInfo.LinkTarget matches the specified expected path.
+        /// If the current platform is Windows and <paramref name="actual"/> is absolute, this method asserts that <paramref name="actual"/> starts with \\?\.
+        /// </summary>
+        protected static void AssertLinkTargetEquals(string expected, string actual)
+        {
+#if WINDOWS
+            if (Path.IsPathFullyQualified(actual))
+            {
+                actual.StartsWith(ExtendedPrefix);
+                expected = Path.Join(ExtendedPrefix, expected);
+            }
+
+            // Windows syscalls remove the redundant segments in the link target path.
+            // We will remove them from the expected path when testing Windows but keep them when testing Unix, which doesn't remove them.
+            int rootLength = PathInternal.GetRootLength(expected);
+            if (rootLength > 0)
+            {
+                expected = PathInternal.RemoveRelativeSegments(expected, rootLength);
+            }
+#endif
+            Assert.Equal(expected, actual);
+        }
+
+
+        /// <summary>
+        /// Asserts that the FullPath of the FileSystemInfo returned by ResolveLinkTarget() matches with the expected path of the file created.
+        /// Trims the Windows device prefix, in case there's any, before comparing.
+        /// </summary>
+        private static void AssertFullNameEquals(string expected, string actual)
+        {
+#if WINDOWS
+            if (PathInternal.IsExtended(actual))
+            {
+                Assert.StartsWith(ExtendedPrefix, actual);
```

Comment:
same as above

Reply:
I think we should keep this, regardless of no longer returning NT paths, `GetFinalPathNameByHandleW` still appends the extended Dos prefix (\\?\). Do you think that we should get rid of that one as well? 
One suggestion from Jeremy K. was to keep the prefix only when the users provides us an extended path; otherwise, remove it.